src/api-object.ts

```diff
@@ -45,7 +45,29 @@ export interface GroupVersionKind {
   readonly kind: string;
 }
 
+const API_OBJECT_SYMBOL = Symbol.for('cdk8s.ApiObject');
+
 export class ApiObject extends Construct {
+
+  /**
+   * Return whether the given object is an `ApiObject`.
+   *
+   * We do attribute detection since we can't reliably use 'instanceof'.
+
+   * @param o The object to check
+   */
+  static isApiObject(o: unknown): o is ApiObject {
+    return o != null && typeof o === 'object' && API_OBJECT_SYMBOL in o;
+  }
+
+  /**
+   * Implements `instanceof ApiObject` using the more reliable `ApiObject.isApiObject` static method
+   *
+   * @param o The object to check
+   */
+  static [Symbol.hasInstance](o: unknown) {
```

Comment:
This is great! TIL.

Would you mind also implementing this for the [`Chart`](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/chart.ts) class? Its also being [used](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/app.ts#L124) with `instanceof`, although it does have a `Chart.isChart` function...